### PR TITLE
RIOT-OS Submodule : Update to 2022.07 release

### PR DESCRIPTION
**Pull request by akshaim**
Update the submodule version. Apparently, release managers forget about this. 

Disclaimer I could not test the existing applications. I wanted to update the submodule so that I can use the latest boards.

### CODE REVIEW

**Comment by akshaim** [comment]
ping @miri64

**Comment by MrKevinWeiss** [comment]
Maybe we can do this automatically?  One less thing for the release manager to worry about.

**Comment by miri64** [comment]
> Maybe we can do this automatically? One less thing for the release manager to worry about.

At least opening a PR should (that then runs the tests introduced in #79), should be doable with the RIOT-release-manager script

**Comment by MrKevinWeiss** [comment]
It seems like it is not the correct commit, at least when I checked...

```diff
diff --git a/RIOT b/RIOT
index 81a01df..72b14ac 160000
--- a/RIOT
+++ b/RIOT
@@ -1 +1 @@
-Subproject commit 81a01df5d94729b8d8a2be7ce8e68f9fc7c7b95f
+Subproject commit 72b14ac73c6fc26e1be53a4da5cd17924b9387b3
```

**Comment by chrysn** [comment]
An idea raised in today's coordination meeting was to just move the applications in with the examples -- IMO they're occupying points in the space spanned by what is in the examples already.

For that, it'd be good to have this in (and yes, the revision is incorrect, but the precise version matters little, and if we went through with the move it'd become moot).

I'm taking the liberty to add a few fixes right inside this PR; they make the code buildable again after API changes.

File: sniffer/Makefile [hunk 1]
```diff
@@ -16,6 +16,7 @@ USEMODULE += shell
 USEMODULE += shell_commands
```
Comment (miri64) [inline sniffer/Makefile:16]:
```
USEMODULE += gnrc_netdev_default
```

is deprecated in 2022.07. Please use `netdev_default` instead.

File: spectrum-scanner/Makefile [hunk 2]
```diff
@@ -12,6 +12,7 @@ USEMODULE += gnrc
 USEMODULE += gnrc_netdev_default
```
Comment (miri64) [inline spectrum-scanner/Makefile:12]: Same here
```suggestion
```

**Review by miri64 — COMMENTED** [review]
(and in `coap-chat/Makefile` as well...)

**Comment by miri64** [comment]
I had some diff lying around for my demo that I meant to PR. Maybe it still fits in this PR (it also contains an unrelated change of the sniffer tool to work properly with python 3):

```diff
diff --git a/sniffer/Makefile b/sniffer/Makefile
index 7c2e39a..72a24db 100644
--- a/sniffer/Makefile
+++ b/sniffer/Makefile
@@ -10,13 +10,12 @@ RIOTBASE ?= $(CURDIR)/../RIOT
 # Define modules that are used
 USEMODULE += fmt
 USEMODULE += gnrc
-USEMODULE += gnrc_netdev_default
 USEMODULE += auto_init_gnrc_netif
+USEMODULE += netdev_default
 USEMODULE += shell
 USEMODULE += shell_commands
 USEMODULE += ps
-USEMODULE += xtimer
-USEMODULE += ztimer64_xtimer_compat
+USEMODULE += ztimer64_usec
 
 # Change this to 0 show compiler invocation lines by default:
 QUIET ?= 1
diff --git a/sniffer/main.c b/sniffer/main.c
index 3f48a3b..32d241a 100644
--- a/sniffer/main.c
+++ b/sniffer/main.c
@@ -23,10 +23,11 @@
 #include <stdio.h>
 
 #include "fmt.h"
+#include "net/gnrc.h"
 #include "thread.h"
-#include "xtimer.h"
 #include "shell.h"
-#include "net/gnrc.h"
+#include "shell_commands.h"
+#include "ztimer64.h"
 
 /**
  * @brief   Buffer size used by the shell
@@ -62,7 +63,7 @@ void dump_pkt(gnrc_pktsnip_t *pkt)
             pkt = gnrc_pktbuf_remove_snip(pkt, pkt->next);
         }
     }
-    uint64_t now = xtimer_now64();
+    uint64_t now_us = ztimer64_now(ZTIMER64_USEC);
 
     print_str("rftest-rx --- len ");
     print_u32_hex((uint32_t)gnrc_pkt_len(pkt));
diff --git a/sniffer/tools/sniffer.py b/sniffer/tools/sniffer.py
index 98b9600..8f44868 100755
--- a/sniffer/tools/sniffer.py
+++ b/sniffer/tools/sniffer.py
@@ -56,10 +56,10 @@ DEFAULT_BAUDRATE = 115200
 def configure_interface(port, channel):
     line = ""
     iface = 0
-    port.write(b'ifconfig\n')
+    port.write('ifconfig\n'.encode())
     while True:
         line = port.readline()
-        if line == '':
+        if line == b'':
             print("Application has no network interface defined",
                   file=sys.stderr)
             sys.exit(2)
```

**Comment by chrysn** [comment]
I've pushed the deprecated module changes. The sniffer changes probably conflict with #61, leaving them as is.

File: spectrum-scanner/Makefile [hunk 3]
```diff
@@ -9,9 +9,10 @@ RIOTBASE ?= $(CURDIR)/../RIOT
 
 # Define modules that are used
 USEMODULE += gnrc
-USEMODULE += gnrc_netdev_default
+USEMODULE += netdev_default
 USEMODULE += auto_init_gnrc_netif
 USEMODULE += xtimer
+USEMODULE += ztimer64_xtimer_compat
```
Comment (miri64) [inline spectrum-scanner/Makefile:15]: Why not use `ztimer64` outright?

Comment (chrysn) [reply spectrum-scanner/Makefile:15]: Works just as well, now using that.

**Comment by miri64** [comment]
Mhhh for some reason the sniffer application does not list an interface anymore with the commit the RIOT submodule currently points to :-/.

File: spectrum-scanner/Makefile (hunk 3 above)
Comment (miri64) [reply spectrum-scanner/Makefile:15]: Mhhh I accidentally commented on the wrong file, but you fixed it for the correct one ^^, but I guess the same question is true for `spectrum-scanner` too.

**Comment by miri64** [comment]
Tested all applications, they all still seem to work (not 100% sure about the spectrum-scanner, as it seems to output always ~90% at my location for all channels).

**Review by miri64 — APPROVED** [review]

**Pull request merged**